diff -ruN libtasn1-4.21.0/configure libtasn1-4.21.0-banan_os/configure
--- libtasn1-4.21.0/configure	2026-01-08 14:19:22.000000000 +0200
+++ libtasn1-4.21.0-banan_os/configure	2026-05-22 20:25:36.208611165 +0300
@@ -9198,6 +9198,10 @@
   lt_cv_deplibs_check_method=pass_all
   ;;
 
+banan_os*)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
+
 beos*)
   lt_cv_deplibs_check_method=pass_all
   ;;
@@ -14911,6 +14915,16 @@
   esac
   ;;
 
+banan_os*)
+  version_type=linux # correct to gnu/linux during the next big refactor
+  need_lib_prefix=no
+  need_version=no
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  dynamic_linker="$host_os DynamicLoader.so"
+  shlibpath_var=LD_LIBRARY_PATH
+  ;;
+
 beos*)
   library_names_spec='$libname$shared_ext'
   dynamic_linker="$host_os ld.so"
