diff -ru python-3.14.7/Modules/resource.c python-3.14.7-banan_os/Modules/resource.c
--- python-3.14.7/Modules/resource.c	2026-08-05 13:29:49.000000000 +0300
+++ python-3.14.7-banan_os/Modules/resource.c	2026-08-25 12:39:57.421597162 +0300
@@ -125,6 +125,7 @@
     PyStructSequence_SetItem(result, 1,
                     PyFloat_FromDouble(doubletime(ru.ru_stime)));
     PyStructSequence_SetItem(result, 2, PyLong_FromLong(ru.ru_maxrss));
+#ifndef __banan_os__
     PyStructSequence_SetItem(result, 3, PyLong_FromLong(ru.ru_ixrss));
     PyStructSequence_SetItem(result, 4, PyLong_FromLong(ru.ru_idrss));
     PyStructSequence_SetItem(result, 5, PyLong_FromLong(ru.ru_isrss));
@@ -138,6 +139,7 @@
     PyStructSequence_SetItem(result, 13, PyLong_FromLong(ru.ru_nsignals));
     PyStructSequence_SetItem(result, 14, PyLong_FromLong(ru.ru_nvcsw));
     PyStructSequence_SetItem(result, 15, PyLong_FromLong(ru.ru_nivcsw));
+#endif
 
     if (PyErr_Occurred()) {
         Py_DECREF(result);
