diff -ruN libmypaint-1.6.1/configure libmypaint-1.6.1-banan_os/configure
--- libmypaint-1.6.1/configure	2020-05-11 19:38:35.000000000 +0300
+++ libmypaint-1.6.1-banan_os/configure	2026-03-25 03:05:53.901886496 +0200
@@ -5608,6 +5608,10 @@
   lt_cv_deplibs_check_method=pass_all
   ;;
 
+banan_os*)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
+
 beos*)
   lt_cv_deplibs_check_method=pass_all
   ;;
@@ -11347,6 +11351,16 @@
   esac
   ;;
 
+banan_os*)
+  version_type=linux # correct to gnu/linux during the next big refactor
+  need_lib_prefix=no
+  need_version=no
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  dynamic_linker="$host_os DynamicLoader.so"
+  shlibpath_var=LD_LIBRARY_PATH
+  ;;
+
 beos*)
   library_names_spec='$libname$shared_ext'
   dynamic_linker="$host_os ld.so"
