diff -ruN AppStream-1.1.2/src/as-system-info.c AppStream-1.1.2-banan_os/src/as-system-info.c
--- AppStream-1.1.2/src/as-system-info.c	2026-01-28 22:15:58.000000000 +0200
+++ AppStream-1.1.2-banan_os/src/as-system-info.c	2026-03-25 04:08:32.249321186 +0200
@@ -513,7 +513,7 @@
 		return 0;
 	}
 	return hbi.memory_size / MB_IN_BYTES;
-#elif defined(__sun)
+#elif defined(__sun) || defined(__banan_os__)
 	long physpages = sysconf (_SC_PHYS_PAGES);
 	long pagesize = sysconf (_SC_PAGESIZE);
 	if (physpages > 0 && pagesize > 0)
